dips/dip-158.md

```diff
@@ -56,21 +58,23 @@ Content:
 * [Appendix A: Sharing Common Payment Identifier and Address](#appendix-a---prerequisite-sharing-common-payment-identifier-and-address)
+* [Appendix B: Include Reference ID in the On-Chain Transaction](#appendix-b---include-reference-id-in-the-on-chain-transaction)
 ---
 ## High-Level Flow and Payment Types
 ---
 The high-level flow of an eCommerce P2M payment is:
 1. On the merchant's checkout page the customer chooses to pay with Diem
 2. The merchant transmits the payment details to its VASP (receiver VASP) so the latter can generate a payment request with Diem  
-3. The the receiver VASP generates a payment request with all details needed to identify this payment later in the process, provides it to the merchant and makes it available to the customer's Diem wallet (the customer's VASP or sender)
-4. The sender VASP parses the payment request (e.g. by scanning the payment request QR) and uses the provided payment identifier to get the payment details. This is the first step of the off-chain protocol
-5. The wallet then presents the requested payment details to the customer so the customer can review the payment
-6. The customer approves the payment request in the wallet
-7. The wallet initializes the payment protocol (off-chain) which will end (if successful) with the funds being charged. Successful completion implies that the receiving VASP was satisfied with the payer information provided and the sender VASP was satisfied with the merchant information provided. **Note** that if the payment amount exceeds the amount that requires a dual attestation signature to be submitted to the blockchain, it must be part of the on-chain transaction submitted  
+3. The receiver VASP generates a payment request with all details needed and provides it to the merchant, so the latter can display the Diem payment details in the checkout page
+4. The customer triggers the payment by performing the required actions in the checkout page, e.g. by scanning a QR code
+5. The sender VASP parses the payment request (e.g. parse the information encoded in the QR code) and uses the provided payment identifier to get the payment details. This is the first step of the off-chain protocol
+6. The wallet then presents the requested payment details to the customer so the customer can review the payment
+7. The customer approves the payment request in the wallet
+8. The wallet initializes the payment protocol (off-chain) which will end (if successful) with the funds being charged and a corresponsing transaction was put on-chain. Successful completion implies that the receiving VASP was satisfied with the payer information provided and the sender VASP was satisfied with the merchant information provided. **Note** that if the payment amount exceeds the amount that requires a dual attestation signature (i.e. above Travel Rule threshold), the receipient signature must be provided to the sender, so the latter can include it in the on-chain transaction submitted.  
```

Comment:
should we use customer VASP and merchant VASP instead of sender VASP and receiver VASP?

Reply:
The glossary section covers that, I think:
>>>
For the sake of simplicity, we use the following terms:
* "Sender", "Payer", "Wallet" or "customer VASP" - The customer's wallet. It is the VASP representing the customer. This is the VASP that will send the funds and eventually will submit the on-chain transaction to the blockchain
* "Receiver", "Receiving VASP" or "Merchant VASP" - The VASP representing the merchant which receives the funds (transaction)
>>>
We previously agreed to use sender/receiver throughout the DIP.